mongo/integration/unified/event_verification.go

```diff
@@ -42,20 +122,26 @@ func verifyEvents(ctx context.Context, expectedEvents *expectedEvents) error {
 		return err
 	}
 
-	if expectedEvents.Events == nil {
-		return nil
+	switch {
+	case expectedEvents.CommandEvents != nil:
+		return verifyCommandEvents(ctx, client, expectedEvents.CommandEvents)
+	case expectedEvents.CMAPEvents != nil:
+		return verifyCMAPEvents(client, expectedEvents.CMAPEvents)
 	}
+	return nil
+}
 
+func verifyCommandEvents(ctx context.Context, client *clientEntity, expectedEvents []commandMonitoringEvent) error {
```

Comment:
This is the same logic that exists in the current implementation of [verifyEvents](https://github.com/mongodb/mongo-go-driver/blob/7ad201b21b690cd7edb8858943115ba7cc1325d4/mongo/integration/unified/event_verification.go#L39). The only new changes are to check the `HasServiceID` fields on the events.